### PR TITLE
Implementation for Java

### PR DESCRIPTION
**Pull request by nexoone**

### CODE REVIEW

File: JavaObject.uno [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		static Matcher<Java.Object> _toJavaObject = new ToJavaObject();
+
+		public static Java.Object FromJSON(Value value)
+		{
+			return value.Match(_toJavaObject);
+		}
+	}
+
+	[ForeignInclude(Language.Java,
+					"java.lang.Object",
```
Comment (ollef) [inline JavaObject.uno]: Something's up with the indentation here.

File: JavaObject.uno [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	                "java.util.Map",
+	                "java.util.HashMap",
+	                "android.util.Log")]	
+	extern(Android) class ToJavaObject : Matcher<Java.Object>
+	{
+		[Foreign(Language.Java)]
+		public Java.Object Case() @{ return null; @}
+
+		[Foreign(Language.Java)]
+		public Java.Object Case(string str) @{ return (String) str; @}
```
Comment (ollef) [inline JavaObject.uno]: This cast is not necessary.

File: JavaObject.uno [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+		[Foreign(Language.Java)]
+		public Java.Object Case(double num) @{ return (double) num; @}
```
Comment (ollef) [inline JavaObject.uno]: Is this cast necessary?

File: JavaObject.uno [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+		[Foreign(Language.Java)]
+		public Java.Object Case(bool b) @{ return (boolean) b; @}
```
Comment (ollef) [inline JavaObject.uno:37]: Is this cast necessary?

File: JavaObject.uno [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+		public Java.Object Case(IEnumerable<KeyValuePair<string, Java.Object>> obj)
+		{
+			debug_log("UNO - ToJavaObject - handle array");
```
Comment (ollef) [inline JavaObject.uno]: Please remove this.

File: JavaObject.uno [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+			var dict = NewHashMap();
+
+			debug_log("UNO - dict: " + dict);
```
Comment (ollef) [inline JavaObject.uno]: Remove.

File: JavaObject.uno [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		[Foreign(Language.Java)]
+		extern(Android) Java.Object NewHashMap()
+		@{
+			debug_log("UNO - NewHashMap");
+
+			return new HashMap<String, Object>();
+		@}
+
+		[Foreign(Language.Java)]
+		extern(Android) void SetDictValue(Java.Object dictHandle, string key, Java.Object value)
```
Comment (ollef) [inline JavaObject.uno]: Consider renaming to e.g. `SetMapValue` to match the Java names better.

File: JavaObject.uno [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+				SetDictValue(dict, keyValue.Key, keyValue.Value);
+			}
+
+			return dict;
+		}
+
+		[Foreign(Language.Java)]
+		extern(Android) Java.Object NewHashMap()
+		@{
+			debug_log("UNO - NewHashMap");
```
Comment (ollef) [inline JavaObject.uno]: Remove.

**Review by ollef — CHANGES_REQUESTED** [review]
Hey, and thanks!

Looks great to me! I added some minor nits as comments.

File: JavaObject.uno (hunk 7 above)
Comment (nexoone) [reply JavaObject.uno]: right...good point

File: JavaObject.uno (hunk 8 above)
Comment (nexoone) [reply JavaObject.uno]: again .... :-)

File: JavaObject.uno (hunk 6 above)
Comment (nexoone) [reply JavaObject.uno]: yes sure

File: JavaObject.uno (hunk 5 above)
Comment (nexoone) [reply JavaObject.uno]: yes sure

File: JavaObject.uno (hunk 4 above)
Comment (nexoone) [reply JavaObject.uno:37]: I don't know .... i do not tested it without the cast

File: JavaObject.uno (hunk 3 above)
Comment (nexoone) [reply JavaObject.uno]: I don't know .... i do not tested it without the cast

File: JavaObject.uno (hunk 2 above)
Comment (nexoone) [reply JavaObject.uno]: correct

File: JavaObject.uno (hunk 1 above)
Comment (nexoone) [reply JavaObject.uno]: ohh ok....

**Comment by ollef** [comment]
Thanks!

**Pull request merged**